File: Modul 4/Soal 4/PRAK404-2410817210010-AryaArrozzaRidhoSyaputra.c
#include <stdio.h>

void jumlah()
{
    float a, b;
    printf("Masukkan nilai pertama: ");
    scanf("%f", &a);
    printf("Masukkan nilai kedua: ");
    scanf("%f", &b);
    float sum = a + b;
    printf("Hasil penjumlahan antara %.2f dengan %.2f adalah %.2f\n\n", a, b, sum);
}

void kurang()
{
    float a, b;
    printf("Masukkan nilai pertama: ");
    scanf("%f", &a);
    printf("Masukkan nilai kedua: ");
    scanf("%f", &b);
    float difference = a - b;
    printf("Hasil pengurangan antara %.2f dengan %.2f adalah %.2f\n\n", a, b, difference);
}

void kali()
{
    float a, b;
    printf("Masukkan nilai pertama: ");
    scanf("%f", &a);
    printf("Masukkan nilai kedua: ");
    scanf("%f", &b);
    float product = a * b;
    printf("Hasil perkalian antara %.2f dengan %.2f adalah %.2f\n\n", a, b, product);
}

void bagi()
{
    float a, b;
    printf("Masukkan nilai pertama: ");
    scanf("%f", &a);
    printf("Masukkan nilai kedua: ");
    scanf("%f", &b);
    float quotient = a / b;
    printf("Hasil pembagian antara %.2f dengan %.2f adalah %.2f\n\n", a, b, quotient);
}

void end()
{
    printf("Terimakasih, telah menggunakan kalkulator Arya Arrozza Ridho Syaputra");
}

void err()
{
    printf("Input anda salah, silahkan coba lagi\n\n");
}

void begin()
{
    printf("Pilih Program\n");
    printf("1. Penjumlahan\n");
    printf("2. Pengurangan\n");
    printf("3. Perkalian\n");
    printf("4. Pembagian\n");
    printf("5. Exit\n");
    printf("Masukkan Pilihan: ");
}

int main()
{
    int prog;

    while (1){
        begin();
        scanf("%d", &prog);
        switch(prog){
        case 1:
            jumlah();
            break;

        case 2:
            kurang();
            break;

        case 3:
            kali();
            break;

        case 4:
            bagi();
            break;

        case 5:
            end();
            return 0;
            break;

        default:
            err();
            break;
        }
    }
    
    return 0;
}

File: Modul 4/Soal 4/PRAK404-2410817210010-AryaArrozzaRidhoSyaputra.py
def jumlah():
    a = float(input("Masukkan nilai pertama: "))
    b = float(input("Masukkan nilai kedua: "))
    sum = a + b
    print(f"Hasil penjumlahan antara {a:.2f} dengan {b:.2f} adalah {sum:.2f}\n")

def kurang():
    a = float(input("Masukkan nilai pertama: "))
    b = float(input("Masukkan nilai kedua: "))
    difference = a - b
    print(f"Hasil pengurangan antara {a:.2f} dengan {b:.2f} adalah {difference:.2f}\n")

def kali():
    a = float(input("Masukkan nilai pertama: "))
    b = float(input("Masukkan nilai kedua: "))
    product = a * b
    print(f"Hasil perkalian antara {a:.2f} dengan {b:.2f} adalah {product:.2f}\n")

def bagi():
    a = float(input("Masukkan nilai pertama: "))
    b = float(input("Masukkan nilai kedua: "))
    quotient = a / b
    print(f"Hasil pembagian antara {a:.2f} dengan {b:.2f} adalah {quotient:.2f}\n")

def end():
    print("Terimakasih, telah menggunakan kalkulator Arya Arrozza Ridho Syaputra")

def err():
    print("Input anda salah, silahkan coba lagi\n")

def begin():
    print("Pilih Program")
    print("1. Penjumlahan")
    print("2. Pengurangan")
    print("3. Perkalian")
    print("4. Pembagian")
    print("5. Exit")
    return input("Masukkan Pilihan: ")

def main():
    while True:
        prog = begin()
        if prog == "1":
            jumlah()
        elif prog == "2":
            kurang()
        elif prog == "3":
            kali()
        elif prog == "4":
            bagi()
        elif prog == "5":
            end()
            break
        else:
            err()

if __name__ == "__main__":
    main()